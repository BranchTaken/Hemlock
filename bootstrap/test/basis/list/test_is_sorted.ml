open! Basis.Rudiments
open! Basis
open List
open Format

let test () =
  let lists = [
    [];
    [0L];
    [0L; 1L];
    [0L; 1L; 2L];

    [0L; 0L];
    [0L; 1L; 1L];
    [0L; 1L; 2L; 2L];

    [1L; 0L];
    [0L; 2L; 1L]
  ] in
  printf "@[<h>";
  iter lists ~f:(fun l ->
    printf "is_sorted               %a -> %b\n"
      (xpp Uns.xpp) l
      (is_sorted l ~cmp:Uns.cmp)
    ;
    printf "is_sorted ~strict:false %a -> %b\n"
      (xpp Uns.xpp) l
      (is_sorted ~strict:false l ~cmp:Uns.cmp)
    ;
    printf "is_sorted ~strict:true  %a -> %b\n"
      (xpp Uns.xpp) l
      (is_sorted ~strict:true l ~cmp:Uns.cmp)
  );
  printf "@]"

let _ = test ()
