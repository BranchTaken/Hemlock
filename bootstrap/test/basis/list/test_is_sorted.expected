is_sorted               [] -> true
is_sorted ~strict:false [] -> true
is_sorted ~strict:true  [] -> true
is_sorted               [0] -> true
is_sorted ~strict:false [0] -> true
is_sorted ~strict:true  [0] -> true
is_sorted               [0; 1] -> true
is_sorted ~strict:false [0; 1] -> true
is_sorted ~strict:true  [0; 1] -> true
is_sorted               [0; 1; 2] -> true
is_sorted ~strict:false [0; 1; 2] -> true
is_sorted ~strict:true  [0; 1; 2] -> true
is_sorted               [0; 0] -> true
is_sorted ~strict:false [0; 0] -> true
is_sorted ~strict:true  [0; 0] -> false
is_sorted               [0; 1; 1] -> true
is_sorted ~strict:false [0; 1; 1] -> true
is_sorted ~strict:true  [0; 1; 1] -> false
is_sorted               [0; 1; 2; 2] -> true
is_sorted ~strict:false [0; 1; 2; 2] -> true
is_sorted ~strict:true  [0; 1; 2; 2] -> false
is_sorted               [1; 0] -> false
is_sorted ~strict:false [1; 0] -> false
is_sorted ~strict:true  [1; 0] -> false
is_sorted               [0; 2; 1] -> false
is_sorted ~strict:false [0; 2; 1] -> false
is_sorted ~strict:true  [0; 2; 1] -> false
